documentation: install instructions with Guix. (#4535)
cocotb/cocotb: docs/source/install.rst

Installation
************

> .. note:: If you want to install the **development version** of cocotb,
> `instructions are here <https://docs.cocotb.org/en/development/install_devel.html>`_.

Reviewer: ```suggestion
.. note::
   If you want to install the **development version** of cocotb,
   `instructions are here <https://docs.cocotb.org/en/development/install_devel.html>`_.
```
This hopefully fixes the unwanted linebreak.

Installation
************

> Using cocotb requires installation of prerequisites and installation of cocotb itself. Alternatively, you may use a package manager.

Reviewer: ```suggestion
Using cocotb requires installation of prerequisites and installation of cocotb itself.
Alternatively, you may use a package manager, see :ref:`install-package-manager`.
```

    Use ``pip -V`` to check.
    If this prints "(python 2.7)", use :command:`pip3` or ``python3 -m pip`` in place of :command:`pip` in the command shown.

.. _install-package-manager:

> Installation using a Package Manager
> ====================================

Reviewer: ```suggestion
Alternative Installation using a Package Manager
================================================
```

The installation instructions vary depending on your package manager:

.. tab-set::

   .. tab-item:: Guix

>       First, setup `Guix-science channel <https://codeberg.org/guix-science/guix-science>`. Then, in a terminal, run

Reviewer: ```suggestion
      First, set up the `Guix-science channel <https://codeberg.org/guix-science/guix-science>`. Then, in a terminal, run
```

      .. code-block:: bash

> 	  guix install python-cocotb # iverilog verilator ghdl-llvm nvc

Reviewer: ```suggestion
	  guix install python-cocotb  # may add any of those: iverilog verilator ghdl-llvm nvc
```
(if that's true?)

Author: Yes

> If you want to install the **development version** of cocotb,

Reviewer: We need to somehow separate this from the section above, right now, it looks like it's part of it: https://cocotb--4535.org.readthedocs.build/en/4535/install.html#installation-using-a-package-manager

Author: I agree. 

I amend my proposal, redirecting readers to the devel version right away with a note at the top: no need to follow remaining instructions if they’re looking for devel instead.

I also include a post install section, common to all installation methods.

    Use ``pip -V`` to check.
    If this prints "(python 2.7)", use :command:`pip3` or ``python3 -m pip`` in place of :command:`pip` in the command shown.

.. _install-package-manager:

Alternative installation using a Package Manager
================================================

The installation instructions vary depending on your package manager:

.. tab-set::

   .. tab-item:: Guix

      First, set up the `Guix-science channel <https://codeberg.org/guix-science/guix-science>`_. Then, in a terminal, run

      .. code-block:: bash

> 	  guix install python-cocotb # may add any of those: iverilog verilator ghdl-llvm nvc

Reviewer 2: The grammar is off. "these"? Personally, I think leaving the phrase out gets the point across with 0 words.

Author: I agree.

Reviewer: I was thinking it could mean that these are the direct dependencies.
